pages/posts/light-clients.md

```diff
@@ -0,0 +1,260 @@
+
+A similar process could also be used to verify state transitions, which would allow light execution clients to verify the validity of the transactions in a block without actually having to execute them. The light client would receive a set of *changes* the were applied between one block and the next and a pair of Merkle proofs for each change (proving the pre-state and the post-state data). This data forms the *witness*. A proof can then be constructed that is much smaller than the witness itself, that demonstrates that the witness, processed according to the Ethereum protocol rules, really geenrates the given post-state. The light client only has to verify the proof to be able to confidently accept the post-state, rather than having to confirm by re-executing transactions as today’s full nodes do.
+
+An extremely lightweight node could be created that only verifies proofs representing the sync committee protocol and state transition function, enablign both consensus and execution to be verified without the need for lots of local storage and transaction execution.
+
+So where's the trade-off? The trade-off is that although verifying a proof is very cheap, *generating* the proof is very expensive. Where verifying a proof may take tens of milliseconds, generating it may take tens of minutes. This is the responsibility of the block-producing full node, adding substantially to an already fairly heavy workload for those nodes. It might eventually require specialized hardware such as GPU arrays or FGPAs. Increasing hardware requirements is generally considered to be a centralizing force.
+
+### Where are we today?
+
+The light clients that exist today are primarily consensus light clients that get their data from RPC requests to a full node and verify against state roots embedded in Beacon block headers. Typically the light client verifies the sync committee signatures, but some newer clients verify SNARKs over the sync committee protocol instead, achieving the same thing but with less computation. Either way, these consensus clients require access to execution data to do anything more interesting than tracking the tip of the chain.
+
+Getting this execution data requires trusting a third party RPC provider to provide honest data without censoring or doing anything nefarious with your information. Some RPC providers expose `eth_getProof` so that you can independently verify the data they provide against a state root from a trusted block header, but some don’t. 
+
+The table below shows which RPC methods are exposed by each of three clients representing a full (execution + Beacon) client, two light clients ([Lodestar’s light client rpc-prover app](https://chainsafe.github.io/lodestar/lightclient-prover/prover/) and [Helios](https://github.com/a16z/helios/blob/master/rpc.md)) and a Portal client (could be Trin, Fluffy or Ultralight - they aim for feature parity). 
```

Comment:
Shameless plug: Nimbus also also has similar client: https://github.com/status-im/nimbus-eth1/tree/master/nimbus_verified_proxy

It uses an integrated version of the the Nimbus consensus light client.